pkg/controller/secretsmanager/secret/setup.go

```diff
@@ -113,6 +121,79 @@ type hooks struct {
 	kube   client.Client
 }
 
+func (e *hooks) lateInitialize(spec *svcapitypes.SecretParameters, resp *svcsdk.DescribeSecretOutput) error {
+	payload, err := e.getPayload(context.TODO(), spec)
+	if err := client.IgnoreNotFound(err); err != nil || payload == nil {
```

Comment:
```suggestion
	if err := client.IgnoreNotFound(err); err != nil || payload != nil {
```
I think my initial code suggestion was the opposite of what you're trying to do in that if statement, sorry about that. We should `return nil` in case payload is full since we don't need to take any action.